src/StakingNode.sol

```diff
@@ -313,43 +337,85 @@ contract StakingNode is IStakingNode, StakingNodeEvents, ReentrancyGuardUpgradea
+    }
+
     //--------------------------------------------------------------------------------------
     //----------------------------------  WITHDRAWALS  -------------------------------------
     //--------------------------------------------------------------------------------------
 
     /**
      * @dev Queues a validator Principal withdrawal for processing. DelegationManager calls EigenPodManager.decreasesShares
-     * which decreases the `podOwner`'s shares by `shares`, down to a minimum of zero.
-     * @param sharesAmount The amount of shares to be queued for withdrawals.
+     * which decreases the `podOwner`'s shares by `depositSharesAmount`, down to a minimum of zero. The actual shares withdrawable
+     * will be less than `depositSharesAmount` depending on the slashing factor
+     * @param depositSharesAmount The amount of deposit shares to be queued for withdrawals.
      * @return fullWithdrawalRoots An array of keccak256 hashes of each withdrawal created.
      */
     function queueWithdrawals(
-        uint256 sharesAmount
-    ) external onlyStakingNodesWithdrawer onlyWhenSynchronized returns (bytes32[] memory fullWithdrawalRoots) {
-        IDelegationManager delegationManager = IDelegationManager(address(stakingNodesManager.delegationManager()));
+        uint256 depositSharesAmount
+    ) external onlyStakingNodesWithdrawer onlyWhenSynchronized returns (bytes32[] memory) {
+
+        IDelegationManager delegationManager = stakingNodesManager.delegationManager();
 
-        IDelegationManager.QueuedWithdrawalParams[] memory params = new IDelegationManager.QueuedWithdrawalParams[](1);
+        IDelegationManagerTypes.QueuedWithdrawalParams[] memory params = new IDelegationManagerTypes.QueuedWithdrawalParams[](1);
         IStrategy[] memory strategies = new IStrategy[](1);
 
         // Assumption: 1 Share of beaconChainETHStrategy = 1 ETH.
         uint256[] memory shares = new uint256[](1);
 
         strategies[0] = beaconChainETHStrategy;
-        shares[0] = sharesAmount;
+        shares[0] = depositSharesAmount;
         // The delegationManager requires the withdrawer == msg.sender (the StakingNode in this case).
-        params[0] = IDelegationManager.QueuedWithdrawalParams({
+        params[0] = IDelegationManagerTypes.QueuedWithdrawalParams({
             strategies: strategies,
-            shares: shares,
-            withdrawer: address(this)
+            depositShares: shares,
+            __deprecated_withdrawer: address(this)
         });
+        uint256 beaconChainSlashingFactor;
+        uint256 withdrawableShares;
+        bytes32[] memory fullWithdrawalRoots;
+        IEigenPodManager eigenPodManager = IEigenPodManager(IStakingNodesManager(stakingNodesManager).eigenPodManager());
+
+        if (delegatedTo == address(0)) {
+            beaconChainSlashingFactor = eigenPodManager.beaconChainSlashingFactor(address(this));
```

Comment:
Please declare variables where they're used

beaconChainSlashingFactor is not used outside of this if block, so is `beaconChainSlashingFactor`